### PR TITLE
ENH Factorization Machine classifier

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds a factorization machine classifier.

I am going to leave the regressor for another PR.

### CODE REVIEW

**Comment by mdagost** [comment]
Oh nice.  I've been looking for a good TF FM implementation :)

**Comment by beckermr** [comment]
Hi @mdagost! We only have a classifier. A PR for the regressor is welcome!

File: muffnn/fm/fm_classifier.py [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+            self._beta = tf.get_variable(
+                "beta", [self.n_dims_, self._output_size])
+            self._beta0 = tf.get_variable("beta0", [self._output_size])
+
+        vx = tf.stack([matmul(self._x, self._v[i, :, :])
+                       for i in range(self.rank)], axis=1)
+        v2 = self._v * self._v
+        v2x2 = tf.stack([matmul(x2, v2[i, :, :])
+                         for i in range(self.rank)], axis=1)
+        int_term = 0.5 * tf.reduce_sum(tf.square(vx) - v2x2, axis=1)
```
Comment (samcarlos) [inline muffnn/fm/fm_classifier.py]: The `axis=1` is a little confusing since its in the middle of this 3-dim array. Maybe change it to `axis=-1` instead?

**Comment by samcarlos** [comment]
Just making sure: there's no way to write eqn 1) here http://www.algo.uni-konstanz.de/members/rendle/pdf/Rendle2010FM.pdf in matrix format, is there?

**Comment by beckermr** [comment]
Yeah equation 1 is the slow version. The TF I have is the best I know of.

**Comment by samcarlos** [comment]
LGTM
